### DISCUSSION TITLE
How to turn off convection term in Oceananigans model

### DISCUSSION THREAD

**Discussion by RWBY-wiess** [CliMA/Oceananigans.jl#2667] [Computational science!] [answered]
I set up an initial buoyancy field, and considered the change law of the flow velocity in all directions under the action of the buoyancy field. For comparison, I need to ignore the convection term in the N-S equation, but don't know how to turn it off in the model.

**Comment by iuryt** [comment]
What about setting `buoyancy=nothing`?
As described in 
https://github.com/CliMA/Oceananigans.jl/blob/main/docs/src/model_setup/buoyancy_and_equation_of_state.md#no-buoyancy

If you don't give any value for the buoyancy argument, it is set to `nothing` as from the example or one-dimensional diffusion.
https://clima.github.io/OceananigansDocumentation/stable/generated/one_dimensional_diffusion/

**Reply by RWBY-wiess** [reply]
thanks~

**Comment by simone-silvestri** [comment, marked as answer]
If by convection term you mean momentum advection in the NS, you can turn it off by specifying `advection = nothing` (in the `NonhydrostaticModel`) and `mometum_advection=nothing` (in the `HydrostaticFreeSurfaceModel` ) and the same to turn it off in the tracer advection-diffusion equation (this time specify `tracer_advection=nothing` in either model)

otherwise if you want to neglect the hydrostatic pressure effects in the NS equations due to the buoyancy field, @iuryt's solution is good! (take care that if you are using the `HydrostaticFreeSurfaceModel` the default buoyancy model is based a linear equation of state so you have to specify `buoyancy=nothing`)

**Reply by RWBY-wiess** [reply]
thanks~
